ngi_visualizations/fpkm_scatter/fpkm_scatter.py

```diff
@@ -26,57 +29,90 @@ def make_fpkm_scatter_plots (input_files, summary=False, output_fn='gene_counts'
     """
     Main function. Takes input files and makes a plot.
     """
-    
-    # Get proper paths for input files
-    input_1 = os.path.realpath(input_files[0])
-    input_2 = os.path.realpath(input_files[1])
-    
-    # First off, assume that we have two FPKM files from cufflinks
-    if summary is False:
-       
-        # Parse the input files
-        sample_1 = load_fpkm_counts(input_1)
-        sample_2 = load_fpkm_counts(input_2)
-        
-        # What are the sample names?
-        sample_1_name = os.path.splitext(os.path.basename(input_1))[0]
-        sample_2_name = os.path.splitext(os.path.basename(input_2))[0]
+    R_dict={}
+    #iterate over all uniqe pairs of input files
+    for f,y in itertools.combinations(input_files,2):
+        # Get proper paths for input files
+        input_1 = os.path.realpath(f)
+        input_2 = os.path.realpath(y)
         
-        # File name
-        if output_fn is None:
-            output_fn = "{}-{}".format(sample_1_name, sample_2_name)
+        # First off, assume that we have two FPKM files from cufflinks
+        if summary is False:
+           
+            # Parse the input files
+            sample_1 = load_fpkm_counts(input_1)
+            sample_2 = load_fpkm_counts(input_2)
+            
+            # What are the sample names?
+            sample_1_name = os.path.splitext(os.path.basename(input_1))[0]
+            sample_2_name = os.path.splitext(os.path.basename(input_2))[0]
+            #print(sample_1_name)
+            #print(sample_2_name)
+            # File name
+            if output_fn is None:
+                output_fn=False
+            # Make the plot
+            plot_filenames = plot_fpkm_scatter(sample_1, sample_2, sample_1_name, sample_2_name, output_fn=output_fn, linear=linear)
+            #Extract the R-value
+            #Get the key from the sample names 
+            fn_key = "{}-{}".format(sample_1_name, sample_2_name) 
```

Comment:
Bear in mind that this will probably fail horribly if we ever have any sample names with hyphens in them ;)